test/YieldStrategyLeverTestBase.sol

```diff
@@ -0,0 +1,303 @@
+        WETH.transfer(address(this), 100000e18);
+
+        vm.prank(timeLock);
+        wethJoin.setFlashFeeFactor(1);
+        vm.prank(timeLock);
+        usdcJoin.setFlashFeeFactor(1);
+        vm.prank(timeLock);
+        daiJoin.setFlashFeeFactor(1);
+    }
+
+    function setUp() public virtual {
+        IPool pool = IPool(ladle.pools(seriesId));
+        FYToken fyToken = FYToken(address(pool.fyToken()));
+        vm.label(address(fyToken), fyToken.symbol());
+        vm.label(address(pool.base()), pool.baseToken().symbol());
+        vm.label(address(pool), "Pool");
+        vm.prank(timeLock);
+        fyToken.setFlashFeeFactor(1);
+
+        lever = new YieldStrategyLever(giver);
+
+        DAI.approve(address(lever), type(uint256).max);
+        USDC.approve(address(lever), type(uint256).max);
+        WETH.approve(address(lever), type(uint256).max);
+
+        AccessControl giverAccessControl = AccessControl(address(giver));
+        giverAccessControl.grantRole(0xe4fd9dc5, timeLock);
+        giverAccessControl.grantRole(0x35775afb, address(lever));
+    }
+
+    /// @notice Create a vault.
+    function leverUp() public returns (bytes12) {
+        vaultId = lever.invest(
+            YieldStrategyLever.Operation.BORROW,
+            seriesId,
+            strategyIlkId, // ilkId edai
+            baseAmount,
+            borrowAmount,
+            fyTokenToBuy,
+            0 //minCollateral,
+        );
+        return vaultId;
+    }
+}
+
+abstract contract Tests is ZeroState {
+    function setUp() public virtual override {
+        super.setUp();
+        vaultId = leverUp();
+    }
+
+    /// @notice Test if vault is created correctly
+    function testVault() public {
+        vaultId = leverUp();
+        DataTypes.Vault memory vault = cauldron.vaults(vaultId);
+        assertEq(vault.owner, address(this));
+
+        assertEq(cauldron.balances(vaultId).art, borrowAmount);
+
+        // Assert that the balances are empty
+        assertEq(
+            IPool(ladle.pools(seriesId)).base().balanceOf(address(lever)),
+            0
+        );
+        assertEq(
+            IPool(ladle.pools(seriesId)).fyToken().balanceOf(address(lever)),
+            0
+        );
+    }
+
+    /// @notice Test if the user is able to repay
+    function testRepay() public {
+        DataTypes.Balances memory balances = cauldron.balances(vaultId);
+        lever.divest(
+            YieldStrategyLever.Operation.REPAY,
+            vaultId,
+            seriesId,
+            strategyIlkId,
+            balances.ink,
+            balances.art,
+            0
+        );
+        balances = cauldron.balances(vaultId);
+        assertEq(balances.ink, 0);
+        assertEq(balances.art, 0);
+        assertEq(
+            IPool(ladle.pools(seriesId)).base().balanceOf(address(lever)),
+            0
+        );
+        assertEq(
+            IPool(ladle.pools(seriesId)).fyToken().balanceOf(address(lever)),
+            0
+        );
```

Comment:
Also test that the user got a profit. You might have to do a few trades in the pool for fees to accumulate and the profit to exist.